docs/authentication.rst

```diff
@@ -422,3 +422,84 @@ Furthermore, as described in the `procedure`_, :envvar:`MAGPIE_USER_REGISTRATION
+and a ``url``. The ``name`` is a the name of that other Magpie instance in the network and should correspond to the
+same value as the :envvar:`MAGPIE_INSTANCE_NAME` value set by the other Magpie instance. The ``url`` is a the root URL
+of the other Magpie instance.
+
+Once a :term:`Network Node` is registered, Magpie can use that other instance to authenticate users as long as the other
+instance also has :envvar:`MAGPIE_NETWORK_MODE` enabled.
+
+Managing Personal JWTs
+~~~~~~~~~~~~~~~~~~~~~~
+
+A :term:`User` can request a new network token with a request to the ``PATCH /token`` route. This route takes one
+optional parameter ``expires`` which is an integer indicating how long (in seconds) until that token expires, the
+default expiry for this token is :envvar:`MAGPIE_DEFAULT_TOKEN_EXPIRY`.
+
+Every time a :term:`User` makes a request to the ``PATCH /token`` route a new token is generated for them. This
+effectively cancels all previously created tokens for that user. If a user wishes to cancel all tokens, they can provide
+an ``expires`` value of ``0`` when making the request.
+
+Authentication
+~~~~~~~~~~~~~~
+
+Once a :term:`User` gets a personal network token, they can use that token to authenticate with any Magpie instance in
+the same network. When a user makes a request, they should set the ``provider_name`` parameter to the value of
+:envvar:`MAGPIE_NETWORK_PROVIDER` and provide the network token in the Authorization header in the following format:
+
+.. code-block:: http
+
+    Authorization: Bearer <network_token>
+
+When using the :ref:`Magpie Adapter <utilities_adapter>`, the token can also be passed as a parameter to the request,
+where the parameter name set by :envvar:`MAGPIE_NETWORK_TOKEN_NAME` and the value is the personal network token.
+
+Authorization
+~~~~~~~~~~~~~
+
+Managing authorization for :term:`Users` who authenticate using personal network tokens is complicated by the fact that
+a :term:`User` is not required to have a full account on both Magpie instances in order to using this authentication
+mechanism. This means that a :term:`User` may be logged in as a node-specific "anonymous" user.
+
+When another Magpie instance is registered as a :term:`Network Node`, a few additional entities are created:
+
+#. a group used to manage the permissions of all users who authenticate using the new :term:`Network Node`.
+   * this group's name will be the :envvar:`MAGPIE_NETWORK_NAME_PREFIX` followed by the :term:`Network Node` name
+#. a group used to manage the permissions of all users who authenticate using *any* other instance in the network
+   * this group's name will be the :envvar:`MAGPIE_NETWORK_GROUP_NAME`
+   * this group will only be created once, when the first :term:`Network Node` is registered
+#. an anonymous user that belongs to the two groups that were just created.
+   * this user name will be the :envvar:`MAGPIE_NETWORK_NAME_PREFIX` followed by the :term:`Network Node` name
```

Comment:
If the `MAGPIE_NETWORK_NAME_PREFIX=pavics` then you shouldn't be able to create any new users whose name starts with `pavics`. Even so, the anonymous user for the `daccs` node would be `pavicsdaccs` (the prefix followed by the network name). 

Now, if you already had a user in the system named `pavicsdaccs` that could be a bit confusing but they should still be able to differentiate them because one would belong to the `pavicsdaccs` group and the original user would not.

If this is easier though, I'm happy to split these users out into different tables as you suggest (https://github.com/Ouranosinc/Magpie/pull/589#pullrequestreview-1627967830) if that will keep things cleaner.

Reply:
Yes, I think that would make it easier conceptually. The API endpoints however would still need to consider these conditions to allow user names or not. I think you mostly already did it with the regexes, but tests must be done to make sure it works everywhere (trying to purposely cause conflict, edge cases and such).

Reply:
Yeah I have some tests on the way but they're still in progress :)